ebu_tt_live/bindings/__init__.py

```diff
@@ -580,24 +580,29 @@ class LiveStyledElementMixin(StyledElementMixin):
 # removed from version 1.0 of the specification.
 
 class RoleMixin(object):
-    _computed_roles = None
+    _computed_roles = set()
+
     def _semantic_compute_roles(self, dataset):
+        current_roles = set()
         if self.role is not None:
-            self._computed_roles = self.role
-        else:
-            if dataset.get('metadata_roles') is not None and len(dataset['metadata_roles']) > 0:
-                self._computed_roles = dataset['metadata_roles'][-1]
-            else:
-                self._computed_roles = None
+            current_roles.update(self.role)
+
+        if dataset.get('role_stack') and dataset['role_stack']:
+            current_roles.update(dataset['role_stack'][-1])
```

Comment:
After this line I'd expect a similar one to pull in any roles defined by the child `<metadata>` element too, e.g. from `dataset['metadata_roles']`.

Reply:
hmm, I got what you mean , updating